app/src/main/java/com/android/shelfLife/viewmodel/InvitationViewModel.kt

```diff
@@ -0,0 +1,109 @@
+package com.android.shelfLife.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import com.android.shelfLife.model.household.HouseHold
+import com.android.shelfLife.model.invitations.Invitation
+import com.android.shelfLife.model.invitations.InvitationRepository
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class InvitationViewModel(private val invitationRepository: InvitationRepository) : ViewModel() {
+
+  private val _invitations = MutableStateFlow<List<Invitation>>(emptyList())
+  val invitations: StateFlow<List<Invitation>> = _invitations.asStateFlow()
+
+  init {
+    FirebaseAuth.getInstance().addAuthStateListener { firebaseAuth ->
+      if (firebaseAuth.currentUser != null) {
+        startListeningForInvitations()
+      } else {
+        stopListeningForInvitations()
+      }
+    }
+  }
+
+  /** Starts listening for real-time updates to invitations. */
+  private fun startListeningForInvitations() {
+    invitationRepository.addInvitationListener(
+        onUpdate = { invitationList -> _invitations.value = invitationList },
```

Comment:
this is perfect, now there is a connection between viewmodel, cache and firebase